### ISSUE TITLE
The cli option --exclude doesn't work  with absolute paths

### ISSUE THREAD

**Issue by Wtyd** [phpmd/phpmd#888]
- PHPMD version: 2.10.0
- PHP Version: 7.4.16
- Installation type: composer
- Operating System / Distribution & Version: Debian 10

### Current Behavior
I use --exclude option and the execution ignore it. The path to source and the path to exclude are absolute. If the exclude path is relative the behavior is right.

### Expected Behavior
The file in the exclude path must be ignored.

### Steps To Reproduce:
1. Create a file with any violation.
2. Execute phpmd with the violated rule against the file (or the path of the file) with the option --exclude in absolute path.
3. Enjoy! 
```bash
# Doesn't Work
vendor/bin/phpmd /var/www/html2/tests/../testsDir/src/File.php ansi unusedcode --exclude "/var/www/html2/tests/../testsDir/src/File.php"

FILE: /var/www/html2/testsDir/src/File.php
------------------------------------------
 21 | VIOLATION | Avoid unused local variables such as '$myVariable'.

Found 1 violation and 0 errors in 109ms

# Works
vendor/bin/phpmd /var/www/html2/tests/../testsDir/src/File.php ansi unusedcode --exclude "testsDir/src/File.php"                                                                    

Found 0 violations and 0 errors in 53ms

No mess detected
```

## Checks before submitting
* [X ] Be sure that there isn't already an issue about this. See: [Issues list](https://github.com/phpmd/phpmd/issues)
* [X ] Be sure that there isn't already a pull request about this. See: [Pull requests](https://github.com/phpmd/phpmd/pulls)
* [ X] I have added every step to reproduce the bug.
* [X ] If possible I added relevant code examples.
* [ X] This issue is about 1 bug and nothing more.
* [ X] The issue has a descriptive title. For example:  "JSON rendering failed on Windows for filenames with space".

**Comment by tvbeek** [comment]
@Wtyd thanks for the report, this is a nice situation that I personally didn't think about.

I think it is a nice feature to support absolute paths, do you have the option to create a PR?

**Comment by Wtyd** [comment]
I've downloaded the project, taken a look at it, and really don't know where to start. Sorry I can not be of more help.

**Comment by AJenbo** [comment]
You need to properly resolve your path, instead of `/var/www/html2/tests/../testsDir/src/File.php` it should be `/var/www/html2/testsDir/src/File.php`.

Since we do pattern based exclusion it's not feasible to do so with anything but canonical paths, the input is not paths but patterns.
